### PR TITLE
[#29] actions for opening tools from workspace list

### PR DESCRIPTION
**Pull request by psiotwo**
As per title.

### CODE REVIEW

File: src/workspaces/actions.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ const WorkspacesActions = {
     'workspaces/addVocabularySuccess',
     'workspaces/addVocabularyFailure'
   )<AddVocabularyPayload, AddVocabularyPayload, Error>(),
+  openInTool: createAction('workspaces/openInToolRequest' )<void>(),
```
Comment (karelklima) [inline src/workspaces/actions.ts]: I'd consider just using the name `workspaces/openInTool`, as the `Request` suffix is usually reserved for the async actions. Also the `<void>` part can be omitted.

File: src/workspaces/components/Tools.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return <>
+        {tools.map( t => <Button
+            onClick={handleClick}
+            color="secondary"
+            href={t.url + "?workspace=" + props.workspaceUri}
+        >{t.label}</Button> )}
+    </>
+}
+
+export default Tools
```
Comment (karelklima) [inline src/workspaces/components/Tools.tsx:20]: There is a code style convention used in the whole repo (basically using Prettier), but I see now that it's not set up as a dependency / executable script... So I will deal with that next week I guess.

File: src/workspaces/components/Tools.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+type ToolProps = { workspaceUri: string }
+
+const Tools: React.FC<ToolProps> = (props: ToolProps) => {
+    const tools = useSelector(getTools);
+    const handleClick = (e : React.MouseEvent) => { e.stopPropagation() };
+    return <>
+        {tools.map( t => <Button
+            onClick={handleClick}
+            color="secondary"
+            href={t.url + "?workspace=" + props.workspaceUri}
```
Comment (karelklima) [inline src/workspaces/components/Tools.tsx:15]: The "string builder" should be done using a template like:
```javascript
`${t.url}?workspace=${props.workspaceUri}`
```

File: src/workspaces/components/Tools.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import {getTools} from "../selectors";
+import Button from "@material-ui/core/Button";
+
+type ToolProps = { workspaceUri: string }
+
+const Tools: React.FC<ToolProps> = (props: ToolProps) => {
+    const tools = useSelector(getTools);
+    const handleClick = (e : React.MouseEvent) => { e.stopPropagation() };
+    return <>
+        {tools.map( t => <Button
```
Comment (karelklima) [inline src/workspaces/components/Tools.tsx:12]: I believe there is a `key` property missing, React should throw a warning in this case

File: src/workspaces/components/Tools.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import {useSelector} from "react-redux";
+import {getTools} from "../selectors";
+import Button from "@material-ui/core/Button";
+
+type ToolProps = { workspaceUri: string }
+
+const Tools: React.FC<ToolProps> = (props: ToolProps) => {
+    const tools = useSelector(getTools);
+    const handleClick = (e : React.MouseEvent) => { e.stopPropagation() };
```
Comment (karelklima) [inline src/workspaces/components/Tools.tsx:10]: I don't think that `onClick` prop on Button is needed if there is `href`

File: src/workspaces/reducers.ts [hunk 6]
```diff
@@ -21,6 +24,7 @@ const initialState: WorkspacesState = {
   isDeleteWorkspaceFormOpen: false,
   isAddVocabularyFormOpen: false,
   workspaces: {},
+  tools: zipObject(tools.map( t => t.url ), tools)
```
Comment (karelklima) [inline src/workspaces/reducers.ts:27]: I can't see a reason of putting the tools into the store... perhaps to make them configurable in the future? I guess it's fine now.

Comment (psiotwo) [reply src/workspaces/reducers.ts:27]: Yeah exactly. A bit overkill now, but was thinking about some configuration of the tools in the future.

File: src/workspaces/components/Tools.tsx (hunk 5 above)
Comment (psiotwo) [reply src/workspaces/components/Tools.tsx:10]: There is onRowClick event for the whole row - without it, the row action redirects to the workspace before forwarding to the external URL, so there is a strange visual effect.

**Pull request merged**